public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/MetricEditor.tsx

```diff
@@ -41,16 +41,16 @@ const isBasicAggregation = (metric: MetricAggregation) => !metricAggregationConf
 
 const getTypeOptions = (
   previousMetrics: MetricAggregation[],
-  esVersion: string,
+  isEs7: string,
```

Comment:
Do you think there is a chance that in near future we will have metrics/features only available in ES 8,9,...?

Reply:
i'm not sure, but i see two arguments for going with this approach:
1. i think next major versions will not happen that fast, so my guess would be this single function stays here as special until we stop supporting es7, and then we can remove it.
2. these attributes are not persisted, so we can change them any time we want, and we can go back to full-semver, if needed.

but i don't have strong feelings there, we can go with full-semver too.

WDYT?

Reply:
still, on further thought, i opted for keeping the old logic with `esVersion` to have a smaller diff. we can always switch to `isEs7` in the future, if we think it makes sense.